crates/store/re_types/src/archetypes/capsules3d_ext.rs

```diff
@@ -0,0 +1,99 @@
+use crate::components;
+
+use super::Capsules3D;
+
+impl Capsules3D {
+    /// Creates a new [`Capsules3D`] with the given axis-aligned lengths and radii.
+    ///
+    /// For multiple capsules, you should generally follow this with
+    /// [`Capsules3D::with_translations()`] and one of the rotation methods, in order to move them
+    /// apart from each other.
+    pub fn from_lengths_and_radii(
+        lengths: impl IntoIterator<Item = impl Into<components::Length>>,
+        radii: impl IntoIterator<Item = f32>,
+    ) -> Self {
+        Self::new(lengths, radii)
+    }
+
+    /// Creates a new [`Capsules3D`] where each capsule extends between the given pairs of points.
+    #[cfg(feature = "glam")]
+    pub fn from_endpoints_and_radii(
+        start_points: impl IntoIterator<Item = impl Into<components::Position3D>>,
+        end_points: impl IntoIterator<Item = impl Into<components::Position3D>>,
+        radii: impl IntoIterator<Item = f32>,
+    ) -> Self {
+        use itertools::Itertools as _; // for .multiunzip()
+
+        let (lengths, translations, quaternions): (
+            Vec<components::Length>,
+            Vec<components::PoseTranslation3D>,
+            Vec<components::PoseRotationQuat>,
+        ) = start_points
+            .into_iter()
+            .zip(end_points)
+            .map(|(p1, p2)| {
+                let p1: glam::Vec3 = p1.into().0.into();
+                let p2: glam::Vec3 = p2.into().0.into();
+
+                // Convert the pair of points to the distance between them and the rotation
+                // from +Z to the direction between them.
+                let direction = p2 - p1;
+                let length = direction.length();
+
+                let quaternion = if length > 0.0 {
```

Comment:
this is more robust against small non-zero lengths (e.g. denormals):

```suggestion
                let direction = direction.normalize_or_zero();

                let quaternion = if direction != glam::Vec3::ZERO {
```